libraries/standard/http/src/http_client.c

```diff
@@ -1934,11 +1942,35 @@ HTTPStatus_t HTTPClient_Send( const TransportInterface_t * pTransport,
                     ( unsigned long ) ( pRequestHeaders->headersLen ) ) );
         returnStatus = HTTP_INVALID_PARAMETER;
     }
+    else if( pRequestHeaders->headersLen > pRequestHeaders->bufferLen )
+    {
+        LogError( ( "Parameter check failed: pRequestHeaders->headersLen > "
+                    "pRequestHeaders->bufferLen." ) );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( pRequestHeaders->headersLen > INT32_MAX )
+    {
+        LogError( ( "Parameter check failed: pRequestHeaders->headersLen > "
+                    "2147483647 (INT32_MAX)." ) );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+    else if( reqBodyBufLen > INT32_MAX )
+    {
+        LogError( ( "Parameter check failed: reqBodyBufLen > "
+                    "2147483647 (INT32_MAX)." ) );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
     else if( ( pResponse != NULL ) && ( pResponse->pBuffer == NULL ) )
     {
         LogError( ( "Parameter check failed: pResponse->pBuffer is NULL." ) );
         returnStatus = HTTP_INVALID_PARAMETER;
     }
+    else if( ( pResponse != NULL ) && ( pResponse->bufferLen > INT32_MAX ) )
```

Comment:
Where do overflow? Why does the bufferLen need to be less than INT32_MAX?

Reply:
Unfortunately, it is only needed for the assert checking if more bytes are received than expected.

Reply:
We resolved this by changing the assert check